exporters/ostream/src/metric_exporter.cc

```diff
@@ -14,54 +14,46 @@ namespace exporter
 namespace metrics
 {
 
+template <typename Container>
+inline void printVec(std::ostream &os, Container &vec)
+{
+  using T = typename std::decay<decltype(*vec.begin())>::type;
+  os << '[';
+  if (vec.size() > 1)
+  {
+    std::copy(vec.begin(), vec.end(), std::ostream_iterator<T>(os, ", "));
+  }
+  os << ']';
+}
+
 OStreamMetricExporter::OStreamMetricExporter(std::ostream &sout) noexcept : sout_(sout) {}
 
 sdk::common::ExportResult OStreamMetricExporter::Export(
-    const nostd::span<std::unique_ptr<opentelemetry::sdk::metrics::MetricData>> &records) noexcept
+    const sdk::metrics::MetricData &data) noexcept
```

Comment:
This may need further changes, based on the review for #1299, as it adds Resources and Instrumentation Info to Metric Data to be exported.

Reply:
#1299 is merged. you can modify the PR to use ResourceMetrics instead of MetricData.

Reply:
@esigo - Just want to check if you want this to be reviewed, as I was thinking to review it once it incorporates changes for ResourceMetrics as mentioned above :)

Reply:
Hi @lalitb, I'm rewriting the unit tests, should be done by tomorrow. I'll ping you when I'm done. 
Thanks

Reply:
Thanks, @esigo. Just wanted to ensure that I am not delaying the review comments in case you are expecting it to be reviewed :)

Reply:
@lalitb this should be ready for review, thanks :)